vllm/model_executor/layers/fused_moe/pplx_prepare_finalize.py

```diff
@@ -6,33 +6,74 @@
 import torch
 
 import vllm.model_executor.layers.fused_moe.modular_kernel as mk
+from vllm.model_executor.layers.fused_moe.config import FusedMoEQuantConfig
 from vllm.model_executor.layers.fused_moe.utils import (
     moe_kernel_quantize_input)
+from vllm.utils import cdiv, round_up
+
+
+def pplx_hidden_dim_scale_bytes(
+    max_num_tokens: int,
+    hidden_dim: int,
+    in_dtype: torch.dtype,
+    quant_dtype: Optional[torch.dtype],
+    per_act_token_quant: bool,
+    block_shape: Optional[list[int]],
+):
+    # For blocked per token: set to
+    #   ceil_div(hidden_dim, block_size) * sizeof(float32)
+    # For per-token: set to 4 * sizeof(float32) (x4 for alignment)
+    if quant_dtype is not None:
+        assert quant_dtype.itemsize == 1
+        hidden_dim_bytes = hidden_dim * quant_dtype.itemsize
+        elem_size = torch.float32.itemsize
+        align = 16
+
+        if per_act_token_quant:
+            # per-token
+            assert block_shape is None
+            hidden_scale_bytes = round_up(max_num_tokens * elem_size, align)
+        elif block_shape is not None:
+            # per-group
+            block_size = block_shape[1]
+            num_blocks = cdiv(hidden_dim, block_size)
+            hidden_scale_bytes = round_up(num_blocks * elem_size, align)
+        else:
+            # per-tensor
+            # ?
+            hidden_scale_bytes = round_up(elem_size, align)
```

Comment:
This must be per token + some tweaks for alignment issues -- @ElizaWszola could you fill in what this comment should say?